### PR TITLE
Adds traceable mod (Ugly icon tho)

### PR DESCRIPTION
**Pull request by el-dasher**
Demo? https://youtu.be/gUFsUK6s9do

### CODE REVIEW

**Comment by el-dasher** [comment]
There are some thjngs that need to be addressed per example saving replays with this mod? isk how to do it later i will try it.

**Comment by Rian8337** [comment]
Perhaps give a short summary about what the mod does or provide a demo. This is important to give people a glance without having to read the code (the game's bad source code makes it worse).

**Comment by el-dasher** [comment]
> Perhaps give a short summary about what the mod does or provide a demo. This is important to give people a glance without having to read the code (the game's bad source code makes it worse).

~~where~~

**Review by Rian8337 — CHANGES_REQUESTED** [review]
1. Since this is a new mod, it should stay unranked for the time being--for reference, you can look at `ScoringScene.load()`:
https://github.com/osudroid/osu-droid/blob/e59e768f89e5d4badead29e4ec42b54abced6fc0/src/ru/nsu/ccfit/zuev/osu/scoring/ScoringScene.java#L481-L512
2. This mod should award silver S (SH/XH) instead of regular S/SS as this can be considered a visual mod, much like Hidden and Flashlight.

File: src/ru/nsu/ccfit/zuev/osu/game/HitCircle.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import org.anddev.andengine.entity.scene.Scene;
 import org.anddev.andengine.entity.sprite.Sprite;
 import org.anddev.andengine.opengl.texture.region.TextureRegion;
+import org.anddev.andengine.util.Debug;
```
Comment (Rian8337) [inline src/ru/nsu/ccfit/zuev/osu/game/HitCircle.java:11]: Perhaps you forgot to remove from testing? 😛

Comment (el-dasher) [reply src/ru/nsu/ccfit/zuev/osu/game/HitCircle.java:11]: Yep

**Comment by el-dasher** [comment]
I Just did that to fix any merge conflicts by now.

**Comment by Rian8337** [comment]
50 files changed...

Not sure how adding TC mod would cause such a large diff?

**Comment by ammarasyad** [comment]
> 50 files changed now...
> 
> Not sure how adding TC mod would cause such a large diff?

Took a quick glance at the files changed tab, it seems that he merged the new commits that was pushed to the main branch. I think he merged the new commits to the wrong branch...

**Comment by kairusds** [comment]
Wouldn't it be better if this was a feature instead of a mod? I think this will be very useful for replays.

**Comment by Beyley** [comment]
> Wouldn't it be better if this was a feature instead of a mod? I think this will be very useful for replays.

what? it's a port of a lazer mod, and how would it "be useful for replays"?

**Comment by kairusds** [comment]
> > Wouldn't it be better if this was a feature instead of a mod? I think this will be very useful for replays.
> 
> what? it's a port of a lazer mod, and how would it "be useful for replays"?

Oh my bad.

**Pull request closed without merging**